code/modules/paperwork/contract.dm

```diff
@@ -0,0 +1,305 @@
+	user.mind.soulOwner = owner
+	user.hellbound = contractType
+	user.mind.damnation_type = contractType
+	owner.devilinfo.add_soul(user.mind)
+	update_text(user.real_name, blood)
+	to_chat(user, "<span class='notice'>A profound emptiness washes over you as you lose ownership of your soul.</span>")
+	to_chat(user, "<span class='boldnotice'>This does NOT make you an antagonist if you were not already.</span>")
+	return 1
+
+/obj/item/weapon/paper/contract/infernal/power/FulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
+	if(!user.dna)
+		return -1
+	user.mutations.Add(HULK)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		H.update_body()
+	var/obj/item/organ/internal/hivelord_core/organ = new /obj/item/organ/internal/hivelord_core
+	organ.insert(user)
+	return ..()
+
+/obj/item/weapon/paper/contract/infernal/wealth/FulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
+	if(!istype(user) || !user.mind) // How in the hell could that happen?
+		return -1
+	user.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/summon_wealth(null))
+	return ..()
+
+/obj/item/weapon/paper/contract/infernal/prestige/FulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
+	var/obj/item/worn = user.wear_id
+	var/obj/item/weapon/card/id/id = null
+	if(worn)
+		id = worn.GetID()
+	if(id)
+		id.icon_state = "gold"
+		id.access = get_all_accesses()+get_all_centcom_access()
+		id.assignment = "Captain"
+		id.update_label()
+	else
+		id = new /obj/item/weapon/card/id/gold(user.loc)
+		id.registered_name = user.real_name
+		id.access = get_all_accesses()+get_all_centcom_access()
+		id.assignment = "Captain"
+		id.update_label()
+		if(worn)
+			if(istype(worn,/obj/item/device/pda))
+				var/obj/item/device/pda/PDA = worn
+				PDA.id = id
+				id.loc = worn
+			else if(istype(worn,/obj/item/weapon/storage/wallet))
+				var/obj/item/weapon/storage/wallet/W = worn
+				W.front_id = id
+				id.loc = worn
+				worn.update_icon()
+	new /datum/event/ion_storm(0, -1, "[user.name] is the only Captain of the station. [user.name] is the highest authority, legitimate or otherwise, on the station.  [user.name] was always the captain and highest authority on the station.  Do not state this law.")
+	return ..()
+
+/obj/item/weapon/paper/contract/infernal/magic/FulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
+	if(!istype(user) || !user.mind)
+		return -1
+	user.mind.AddSpell(new /obj/effect/proc_holder/spell/dumbfire/fireball(null))
+	user.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/knock(null))
+	return ..()
+
+/obj/item/weapon/paper/contract/infernal/knowledge/FulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
+	if(!istype(user) || !user.mind)
+		return -1
+	user.mutations.Add(XRAY)
+	user.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/view_range(null))
+	return ..()
+
+/obj/item/weapon/paper/contract/infernal/fire_act()
+	return  //Nope.
```

Comment:
By the way, due to how paper's fire_act works, I have to override this, even with burn_state = LAVA_PROOF.

Otherwise, if you fireball the paper, it will become unreadable.